plugins/signalmonitor/signalhistorydelegate.cpp

```diff
@@ -95,10 +95,15 @@ void SignalHistoryDelegate::paint(QPainter *painter, const QStyleOptionViewItem
 
     foreach (qint64 ev, events) {
         const qint64 ts = SignalHistoryModel::timestamp(ev);
-        if (ts >= startTime && ts < endTime) {
-            const int x = x0 + dx * (ts - startTime) / interval;
-            lines << QLine(x, y0 + 1, x, y0 + dy - 2);
-        }
+
```

Comment:
how is this optimizing the rendering? the code is exactly the same as before, you just split the conditoinal in two - this cannot possibly influence the performance?!